### PR TITLE
[ST] enable all system test disable due to UTO and add admin client utils 

### PR DESCRIPTION
**Pull request by henryZrncik**
### Type of change

- Refactoring

### Description

**Allow all tests disabled due to `UTONotSupported`**
refactor of all checks and tests in ST suites to run with UTO. 
- removing some steps from recovery tests and whole test altogether (`TopicST#testCreateTopicViaKafka`)  which is now (with UTO only) doing nothing
- removal of `UTONotSupported` and `UTONotSupportedCondition`

**Some of Kafka Scripts usages transition to Admin Client**. 
- implementing functionality for adminClient command `describeTopic` which can parse KafkaTopicDescription object with `partitionCount`, `name`, `replicaCount` (which are effectively all we care about the kafka topic in tests so far). 
- adding 8 methods into `AdminClientTemplates` for deploying admin client on any of combination `SSL`, `PLAINTEXT`, `SASL_PLAINTEXT` , `SASL_SSL`  security with/without extra `configuration` parameter.  
- removal of some methods from KafkaTopicUtils which works with kafka scripts  which can be now ignored due to using adminClient. 
- add new label into admin client which would resolve problem of having unique labels in case there are 2 admin clients at the same namespace. 

**Minor refactors**
- admin client deployment.
- util methods in cruise control and their calls in Cruise Control STs.

### CODE REVIEW

**Comment by henryZrncik** [comment]
Do not merge before release of new test clients.

File: systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/admin/KafkaAdminException.java
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/admin/KafkaAdminException.java:7]: Javadoc :)

File: systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java:37]:
```suggestion
    //              TLS (SSL)
```

File: systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java:60]: Templates servers as space where you have a definition of all objects, which we use in our test suites (i.e., we have pre-defined resources that repeat many times). Here you break this contract by adding inside `Templates` a ResourceManager, which is bad. You should always use 
```java
AdminClientTemplates.tlsAdminClient(
                namespace,
                userName,
                adminName,
                clusterName,
                bootstrapName,
                additionalConfig
                ));
```
In test case and call `rm.createResourceWithWait(....` also in test case.

File: systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java:118]: same as previous comment

File: systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java:173]: same as previous comment

File: systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java:217]: same as previous comment

File: systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/objects/PodUtils.java
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/objects/PodUtils.java:79]: Why this unnecessary change? In the whole `systemtest` we mostly use `namespaceName` for String based.

File: systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlConfigurationST.java
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlConfigurationST.java:97]: This is what I don't want. Everything related to resources should go directly through ResourceMananger. And the get it via I think as this is deployment you would need to use `DeploymentResource.get...`similarly as we do for KafkaResource.

File: systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java:185]: same as previous

File: systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java:319]: you know what I am gonna say... :D

File: systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java:445]: same

**Review by see-quick — COMMENTED** [review]
Looks good. That's a lot of work done thanks 💯 ! I have a few comments ...

File: systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java:241]: This seems like a generic to use in other resources also.

File: systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java:251]:
```suggestion
    private static List<EnvVar> buildTLSUserCredentials(String userName) {
```

File: systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java:252]:
```suggestion
        EnvVar userCertificate = new EnvVarBuilder()
```

File: systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java:262]:
```suggestion
        EnvVar userPrivateKey = new EnvVarBuilder()
```

File: systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java:131]: Don't we have inside `TestStorage` a method that gets such value? Moreover, we could then use just the TestStorage instance to pass it to `plainAdminClient`. But that's for another PR :)

File: systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java:461]: Wouldn't be better to get such value via fabri8? I mean that magic `3`. Or a better way is to factor out that 3 with local variable :) also in creating i.e.,
```diff
- KafkaTopicTemplates.topic(testStorage.getSourceClusterName(), testStorage.getTopicName(), 3, testStorage.getNamespaceName()).build(),
+ KafkaTopicTemplates.topic(testStorage.getSourceClusterName(), testStorage.getTopicName(), sourceTopicPartitions, testStorage.getNamespaceName()).build(),
 ```

**Review by see-quick — COMMENTED** [review]
Looks good thanks., a few suggestions to consider

File: systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java
Comment (henryZrncik) [reply systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java:241]: there are only few (2) resources that needs something like this and 1 of them are clients which have this provided internally.

File: systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java
Comment (henryZrncik) [reply systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java:131]: not exactly but yeah this seems like good solution in case we will use this extensively.

**Review by see-quick — APPROVED** [review]
LGTM, Assuming that the tests will pass :)

File: systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java
Comment (im-konge) [inline systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java:55]: FMPOV it's a bit strange to have method `tlsAdminClient` which returns modified `plainAdminClient`. Maybe it would make sense to have `defaultAdminClient` as we have for other clients?

File: systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java
Comment (im-konge) [inline systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java:66]:
```suggestion
                .endTemplate()
            .endSpec()
            .build();
    }
```

File: systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java
Comment (im-konge) [inline systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java:100]:
```suggestion
                .endTemplate()
            .endSpec()
            .build();
```

File: systemtest/src/main/java/io/strimzi/systemtest/utils/AdminClientUtils.java
Comment (im-konge) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/AdminClientUtils.java:42]: Do these deserve javadoc as well as others?

**Review by im-konge — COMMENTED** [review]
I left couple of comments, but otherwise this looks good. Thanks a lot :)

File: systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/admin/AdminClient.java
Comment (henryZrncik) [reply systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/admin/AdminClient.java:26]: because this method can be called several times and i did not want to specify it repeatedly; 
also this way it can be configured here for all usages (including future extensions) once only, without need for introducing new methods WDYT ?

File: systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java
Comment (henryZrncik) [reply systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java:55]: good point thanks.

File: systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/admin/AdminClient.java
Comment (im-konge) [reply systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/admin/AdminClient.java:26]: well when you will need it in future, I guess you can add it? We had more things like this, which were prepared for the future, but actually they were never used and after few months/years they were removed. So in case that you will need it in the future, you can do it that way.

Comment (henryZrncik) [reply systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/admin/AdminClient.java]: yeah i see i was mainly focusing on fact that given command (`describe`) can be called several times and we do no need to set that property several times. But Yeah overall for now setting it this way is probably overkill.

Comment (im-konge) [reply systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/admin/AdminClient.java]: From what I saw I think that it wasn't needed, as you anyway have the whole record class for that result, so there will be no unknown properties I suppose. Anyway, thanks

**Review by im-konge — APPROVED** [review]
LGTM, thanks :) good job

**Pull request merged**
